<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Enrico Ros</author>
 <class>InfoView</class>
 <widget class="QWidget" name="InfoView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>572</width>
    <height>407</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Inspector Plugin Information</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="iconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="pixmap">
        <pixmap resource="../inspector.qrc">:/inspector/images/submenu-up.png</pixmap>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="titleLabel">
       <property name="text">
        <string>&lt;b&gt;Information / Status / Debugging&lt;/b&gt; (temporary)</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTabWidget" name="Tabwidget1">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="pluginWidget">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="icon">
       <iconset resource="../inspector.qrc">
        <normaloff>:/inspector/images/inspector-icon-32.png</normaloff>:/inspector/images/inspector-icon-32.png</iconset>
      </attribute>
      <attribute name="title">
       <string>Plugin</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="1" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Plugin Loaded:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="../inspector.qrc">:/inspector/images/status-ok.png</pixmap>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>General Enable:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QToolButton" name="enaButton">
         <property name="iconSize">
          <size>
           <width>10</width>
           <height>10</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+S, Ctrl+R</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Modules:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="modLabel"/>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="2">
        <widget class="QLabel" name="label_13">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>New Instance Activation Options</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Show Painted Areas:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="paintBox">
         <property name="text">
          <string>enable</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Memory Allocation Analysis:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QCheckBox" name="mallocBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>enable</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_16">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Inspector Status</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="instanceWidget">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="icon">
       <iconset resource="../../debugger/debugger.qrc">
        <normaloff>:/debugger/images/debugger_start.png</normaloff>:/debugger/images/debugger_start.png</iconset>
      </attribute>
      <attribute name="title">
       <string>Instance 1</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Program is being debugged:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="debLabel"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Probe Injected:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="injLabel"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Probe Activated:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="actLabel"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Probe Connected:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="conLabel"/>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Working:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="workLabel"/>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="debugWidget">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <attribute name="icon">
       <iconset resource="../../coreplugin/fancyactionbar.qrc">
        <normaloff>:/fancyactionbar/images/mode_Debug.png</normaloff>:/fancyactionbar/images/mode_Debug.png</iconset>
      </attribute>
      <attribute name="title">
       <string>Debug CommServer</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QCheckBox" name="checkBox">
         <property name="text">
          <string>Show debugging</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolBox" name="debugToolBox">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <widget class="QWidget" name="page">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>85</width>
            <height>108</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Service</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPlainTextEdit" name="serviceText"/>
           </item>
           <item>
            <widget class="QToolButton" name="toolButton">
             <property name="text">
              <string>clear</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>85</width>
            <height>108</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Messages</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QPlainTextEdit" name="messagesText"/>
           </item>
           <item>
            <widget class="QToolButton" name="toolButton_2">
             <property name="text">
              <string>clear</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>85</width>
            <height>108</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Errors</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPlainTextEdit" name="errorsText"/>
           </item>
           <item>
            <widget class="QToolButton" name="toolButton_3">
             <property name="text">
              <string>clear</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_4">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>529</width>
            <height>43</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Other</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="progressLabel">
               <property name="text">
                <string>Progress</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QProgressBar" name="progressBar">
               <property name="value">
                <number>0</number>
               </property>
               <property name="invertedAppearance">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_12">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>&lt;b&gt;Usage:&lt;/b&gt; For testing Inspector, &lt;em&gt;start the executable in debugging mode and pause it&lt;/em&gt;. This is when all the tricky work takes place. Of course the &quot;debugging helpers&quot; must be the special ones, built just for this purpose.&lt;br/&gt;&lt;br/&gt;&lt;b&gt;Disclaimer:&lt;/b&gt; Inspector is in beta stage. The concept works, however the implementation is still not complete. In particular be aware of the following limitations:&lt;br/&gt; - the analysis is done on the main thread (gui) events only&lt;br/&gt; - runtime reconfiguration is not allowed&lt;br/&gt; - it's not optimized and there are bugs</string>
     </property>
     <property name="textFormat">
      <enum>Qt::RichText</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../coreplugin/fancyactionbar.qrc"/>
  <include location="../../debugger/debugger.qrc"/>
  <include location="../inspector.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>debugToolBox</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>43</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>41</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
